src/Explorer/Tabs/CloudShellTab/ShellTypes/PostgresShellHandler.tsx

```diff
@@ -0,0 +1,44 @@
+import { userContext } from "../../../../UserContext";
+import { AbstractShellHandler } from "./AbstractShellHandler";
+
+const PACKAGE_VERSION: string = "15.2";
+
+export class PostgresShellHandler extends AbstractShellHandler {
+  private _endpoint: string | undefined;
+
+  constructor() {
+    super();
+    this._endpoint = userContext?.databaseAccount?.properties?.postgresqlEndpoint;
+  }
+
+  public getShellName(): string {
+    return "PostgreSQL";
+  }
+
+  public getSetUpCommands(): string[] {
+    return [
+      "if ! command -v psql &> /dev/null; then echo '⚠️ psql not found. Installing...'; fi",
+      `if ! command -v psql &> /dev/null; then curl -LO https://ftp.postgresql.org/pub/source/v${PACKAGE_VERSION}/postgresql-${PACKAGE_VERSION}.tar.bz2; fi`,
+      `if ! command -v psql &> /dev/null; then tar -xvjf postgresql-${PACKAGE_VERSION}.tar.bz2; fi`,
+      "if ! command -v psql &> /dev/null; then mkdir -p ~/pgsql; fi",
+      "if ! command -v psql &> /dev/null; then curl -LO https://ftp.gnu.org/gnu/readline/readline-8.1.tar.gz; fi",
+      "if ! command -v psql &> /dev/null; then tar -xvzf readline-8.1.tar.gz; fi",
+      "if ! command -v psql &> /dev/null; then cd readline-8.1 && ./configure --prefix=$HOME/pgsql; fi",
+      "if ! command -v psql &> /dev/null; then echo 'export PATH=$HOME/pgsql/bin:$PATH' >> ~/.bashrc; fi",
+      "source ~/.bashrc",
+    ];
+  }
+
+  public getConnectionCommand(): string {
+    if (!this._endpoint) {
+      return `echo '${this.getShellName()} endpoint not found.'`;
+    }
+
+    const loginName = userContext.postgresConnectionStrParams.adminLogin;
+    return `psql -h "${this._endpoint}" -p 5432 -d "${loginName}" -U "${loginName}" --set=sslmode=require`;
```

Comment:
Username and database name are the same ?

Reply:
thanks for pointing it, I reverified it, default database name should be "citus" https://learn.microsoft.com/en-us/azure/cosmos-db/postgresql/reference-limits#database-creation